v03_pipeline/lib/misc/family_loading_failures.py

```diff
@@ -10,83 +12,93 @@ def passes_relatedness_check(
     if other_id is None:
-        return True
+        return True, None
     coefficients = relatedness_check_lookup.get(
         (min(sample_id, other_id), max(sample_id, other_id)),
     )
-    if not coefficients:
-        return False
-    return np.allclose(
+    if not coefficients or not np.allclose(
         coefficients,
         relation.coefficients,
         0.1,
-    )
+    ):
+        return (
+            False,
+            f'Sample {sample_id} has expected relation "{relation.value}" to {other_id} but has coefficients {coefficients or []}',
+        )
+    return True, None
 
 
-def passes_all_relatedness_checks(  # noqa: C901
+def all_relatedness_checks(  # noqa: C901
     relatedness_check_lookup: dict[tuple[str, str], list],
     sample: Sample,
-) -> bool:
+) -> list[str]:
+    failure_reasons = []
     for parent_id in [sample.mother, sample.father]:
-        if not passes_relatedness_check(
+        success, reason = passes_relatedness_check(
             relatedness_check_lookup,
             sample.sample_id,
             parent_id,
             Relation.PARENT,
-        ):
-            return False
+        )
+        if not success:
+            failure_reasons.append(reason)
 
     for grandparent_id in [
         sample.maternal_grandmother,
         sample.maternal_grandfather,
         sample.paternal_grandmother,
         sample.paternal_grandfather,
     ]:
-        if not passes_relatedness_check(
+        success, reason = passes_relatedness_check(
             relatedness_check_lookup,
             sample.sample_id,
             grandparent_id,
             Relation.GRANDPARENT,
-        ):
-            return False
+        )
+        if not success:
+            failure_reasons.append(reason)
 
     for sibling_id in sample.siblings:
-        if not passes_relatedness_check(
+        success, reason = passes_relatedness_check(
             relatedness_check_lookup,
             sample.sample_id,
             sibling_id,
             Relation.SIBLING,
-        ):
-            return False
+        )
+        if not success:
+            failure_reasons.append(reason)
 
     for half_sibling_id in sample.half_siblings:
         # NB: A "half sibling" parsed from the pedigree may actually be a sibling, so we allow those
         # through as well.
-        if not passes_relatedness_check(
+        success1, _ = passes_relatedness_check(
             relatedness_check_lookup,
             sample.sample_id,
             half_sibling_id,
-            Relation.HALF_SIBLING,
-        ) and not passes_relatedness_check(
+            Relation.SIBLING,
+        )
+        success2, reason = passes_relatedness_check(
             relatedness_check_lookup,
             sample.sample_id,
             half_sibling_id,
-            Relation.SIBLING,
-        ):
-            return False
+            Relation.HALF_SIBLING,
+        )
+        if not success1 and not success2:
+            failure_reasons.append(reason)
```

Comment:
a bit turned around my logic here - fail if they pass tests for both sibling and half-sibling, yeah? 

Would there ever be a case where they would fail at being half-siblings but pass at being full siblings?

Reply:
yeah, this is specific to the "half-siblings" parsed from the pedigree (meaning a single parent in common).  the logic here means that we won't fail the family if a half-sibling in the pedigree is actually a full sibling in the ibd.  the relationship is a "success" if either of the checks passes.  

we check the full siblings [here](https://github.com/broadinstitute/seqr-loading-pipelines/pull/704/files/0812e75ecbedd78dbed33b7e23760cb1dceaa4a0#diff-81bda30e9edc96dbaf6b9533d4db17f8afa17df35de4b37c3c05e1856e6779dbR64), where both parents are shared.